frame/support/src/traits.rs

```diff
@@ -2228,6 +2228,23 @@ pub trait GetPalletVersion {
 	fn storage_version() -> Option<PalletVersion>;
 }
 
+/// Something that can execute a given block.
+///
+/// Executing a block means that all extrinsics in a given block will be executed and the resulting
+/// header will be checked against the header of the given block.
+pub trait ExecuteBlock<Block: BlockT> {
+	/// Execute the given `block`.
+	///
+	/// This will execute all extrinsics in the block and check that the resulting header is correct.
+	///
+	/// Returns the result header.
+	///
+	/// # Panic
+	///
+	/// Panics when an extrinsics panics or the resulting header doesn't match the expected header.
+	fn execute_block(block: Block) -> Block::Header;
```

Comment:
doesn't the block already include a header, where it is equal to the one being returned?

Reply:
Yeah you are right. I also thought about this yesterday and than I thought that I had found something that requires this, but after thinking about this again, I do not require to return the header here. :(

Reply:
will we revert?